### PR TITLE
added annotation functions for reports

### PR DESCRIPTION
**Pull request by m-kovalsky**

### CODE REVIEW

File: src/sempy_labs/report/_reportwrapper.py [hunk 1]
```diff
@@ -1712,141 +1820,195 @@ def delete_key_in_json(obj, key_to_delete):
...
                 self, page_name=page_name
             )
+        elif page_name is not None and visual_name is not None:
+            page_name, page_display_name, visual_name, file_path = (
+                helper.resolve_visual_name(
+                    self, page_name=page_name, visual_name=visual_name
+                )
+            )
+        else:
+            raise ValueError(f"{icons.red_dot}")
```
Comment (eisber) [inline src/sempy_labs/report/_reportwrapper.py:1944]: what's the error?

**Review by eisber — CHANGES_REQUESTED** [review]

File: src/sempy_labs/report/_reportwrapper.py [hunk 2]
```diff
@@ -1296,6 +1350,71 @@ def list_report_level_measures(self) -> pd.DataFrame:
...
+            elif path.endswith("/visual.json"):
+                file = _decode_b64(payload)
+                json_file = _load_json(file)
+                page_display = visual_mapping.get(path)[1]
+                visual_name = json_file.get("name")
+                if "annotations" in json_file:
+                    for ann in json_file["annotations"]:
+                        new_data = {
+                            "Type": "Visual",
+                            "Object Name": f"'{page_display}'[{visual_name}]",
```
[the remark this answers is not in the document]
Comment (m-kovalsky) [reply src/sempy_labs/report/_reportwrapper.py:1408]: i don't think so. the visual_mapping function handles retrieving the page_display.

File: src/sempy_labs/report/_reportwrapper.py [hunk 3]
```diff
@@ -1712,141 +1820,195 @@ def delete_key_in_json(obj, key_to_delete):
...
                 )
-            valid_page_name, valid_display_name, visual_name, file_path = (
-                helper.resolve_visual_name(self, page_name=p_name, visual_name=v_name)
             )
-            rd_filt = rd[rd["path"] == file_path]
-            payload = rd_filt["payload"].iloc[0]
-            obj_file = base64.b64decode(payload).decode("utf-8")
-            obj_json = json.loads(obj_file)
+        else:
+            raise ValueError(f"{icons.red_dot}")
```
[the remark this answers is not in the document]
Comment (m-kovalsky) [reply src/sempy_labs/report/_reportwrapper.py]: added

File: src/sempy_labs/report/_reportwrapper.py (hunk 1 above)
Comment (m-kovalsky) [reply src/sempy_labs/report/_reportwrapper.py]: added

**Review by eisber — APPROVED** [review]

**Pull request merged**